syntax = "proto2";

package proto;

message StopEndpoint {
  repeated Stop stop = 1;
}

message RouteEndpoint {
  repeated Route route = 2;
}

message StopDetailEndpoint {
  required Stop stop = 1;
}

message RouteDetailEndpoint {
  required Route route = 1;
}

message Stop {
  required string id = 1;
  required string name = 2;
  required Location location = 3;
  required StopAccessibility accessibility = 4;
}

message StopAccessibility {
  required WheelchairStopAccessibility wheelchair = 1;
}

enum WheelchairStopAccessibility {
  WHEELCHAIR_STOP_ACCESSIBILITY_UNKNOWN = 0;
  WHEELCHAIR_STOP_ACCESSIBILITY_NONE = 1;
  WHEELCHAIR_STOP_ACCESSIBILITY_PARTIAL = 2;
  WHEELCHAIR_STOP_ACCESSIBILITY_FULL = 3;
}

message Route {
  required string id = 1;
  required string code = 2;
  required string name = 3;
  required RouteType type = 4;
}

enum RouteType {
  ROUTE_TYPE_UNKNOWN = 0;
  ROUTE_TYPE_TRAM = 1;
  ROUTE_TYPE_METRO = 2;
  ROUTE_TYPE_RAIL = 3;
  ROUTE_TYPE_BUS = 4;
  ROUTE_TYPE_FERRY = 5;
}

message StopTimetable {
  repeated StopTimetableTime times = 1;
}

message StopTimetableTime {
  required string routeId = 1;
  required string routeCode = 2;
  required string arrivalTime = 3;
  required string departureTime = 4;
  required string heading = 5;
  required int32 sequence = 6;
}

message Location {
  required double lat = 1;
  required double lng = 2;
}